### DISCUSSION TITLE
Implement batch identify feature

### DISCUSSION THREAD

**Discussion by tlammer** [specify/specify7#5526] [Feature Requests]
Michigan would like to be able to add a batch of preparations to existing records.  It would be adding the same preparation information (count, type, etc.) to different collection objects.
As a case study, John said:
They created a bunch of tissue preps to send out as a gift, and now need to record them. The collection manager finds the process of creating all of the preparation records onerous. He feels that Specify can do things to be more efficient to use.

**Comment by ruhfel** [comment]
To add a bit more info here, I requested this feature for the following reasons. 

Right now if you have to add a prep type of "DNA sample" to 50 different collection objects (like I have had to do twice today) so that you can register them as part of a Gift, it is very laborious. 

Once you have the Collection Object Record open, you have to click 9 times to add a DNA Sample Prep Type. If you have to do this for 50 specimens that is 450 clicks, plus some typing for each one.

Then, once you add all these to a gift, you have to individually select the DNA Prep Type 50 different times. This should also be changed so you could select all of one Prep type. As is is now, you have to individually select the prep type for all collection objects. Happy to explain further if needed.

**Comment by maxpatiiuk** [comment]
I believe this can be done through the WorkBench since Specify 7.6
Choose Loan Preparation as a base table and add `Loan > Loan Number` field to the mapping

**Comment by grantfitzsimmons** [comment]
I am able to construct a data set that allows batch uploading of loan preparations. unfortunately, each preparation must have a column that points to the loan number separately.

**Comment by maxpatiiuk** [comment]
@tlammer Does this satisfy the use case?

**Comment by tlammer** [comment]
I believe that does fit the use case. I think a tool like the batch identify tool in 6 would be something to strive for in the future as it is easier for the user (fields are displayed as they are on the form so the user doesn't have to worry about mapping correctly, and they only have to enter the data once and don't have to worry about duplicating the data for each CO they want to add it to) but for a first round, it meets the requirements

**Comment by maxpatiiuk** [comment]
Will be included in the batch editor

**Comment by grantfitzsimmons** [comment]
Arguably already done using the WorkBench
